plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergOrcConnectorTest.java

```diff
@@ -45,4 +53,36 @@ protected Session withSmallRowGroups(Session session)
                 .setCatalogSessionProperty("iceberg", "orc_writer_max_stripe_rows", "10")
                 .build();
     }
+
+    @Test
+    public void testTinyintType()
+            throws Exception
+    {
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_tinyint", "(\"_col0\" int)")) {
+            // Insert dummy rows and replace the ORC file that the type is struct<_col0:smallint>
+            // Avoid inserting rows using TestTable to generate only 1 file
+            assertUpdate("INSERT INTO " + table.getName() + " VALUES (-127), (NULL)", 2);
+            Path orcFilePath = Path.of((String) computeScalar(format("SELECT DISTINCT file_path FROM \"%s$files\"", table.getName())));
+            String checksumFilePath = format("%s/.%s.crc", orcFilePath.getParent(), orcFilePath.getFileName());
+            Files.copy(Path.of(Resources.getResource("single_tinyint_column/data.orc").getPath()), orcFilePath, REPLACE_EXISTING);
```

Comment:
- single_tinyint_column/data.orc ->  single-tinyint-column.orc
- `Path.of(url.getPath())` is not the correct way to convert from URL to a local file. see #10086